### PR TITLE
libkdumpfile: abstract out dependencies and sync with upstream

### PR DESCRIPTION
**Pull request by sdimitro**
Relevant libkdumpfile PR: https://github.com/delphix/libkdumpfile/pull/1

## Testing

Created userland pkg-build job setting the following:
```
LINUX_PKG_GIT_URL: https://github.com/sdimitro/linux-pkg.git
LINUX_PKG_GIT_BRANCH: libkdumpfile_sync
SINGLE_PACKAGE_GIT_URL: https://github.com/sdimitro/libkdumpfile.git
SINGLE_PACKAGE_GIT_BRANCH: debian_dir
```
URL: http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/linux-pkg-build/job/master/job/userland/job/pre-push/527/

### CODE REVIEW

File: packages/libkdumpfile/config.sh [hunk 1]
```diff
@@ -39,3 +32,7 @@ function build() {
 	# Install libkdumpfile, it's needed to build drgn
 	logmust install_pkgs "$WORKDIR/artifacts"/*.deb
 }
+
+function update_upstream() {
+	logmust update_upstream_from_git
```
Comment (pzakha) [inline packages/libkdumpfile/config.sh]: You need to also include libkdumpfile in `package-lists/update/userland.pkgs` for this to take effect.

Note that you'll also need to create an `upstreams/master` branch in https://github.com/delphix/libkdumpfile that points to the latest commit from upstream.

Comment (sdimitro) [reply packages/libkdumpfile/config.sh]: Created the branch in the libkdumpfile repo : https://github.com/delphix/libkdumpfile/tree/upstreams/master

And I also updated `userland.pkgs`.

**Comment by sdimitro** [comment]
Reran the userland pre-push after the above changes -> http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/linux-pkg-build/job/master/job/userland/job/pre-push/528/

**Comment by sdimitro** [comment]
After adding `dh-python` to list of deps for `libkdumpfile` -> http://selfservice.jenkins.delphix.com/job/devops-gate/job/master/job/linux-pkg-build/job/master/job/userland/job/pre-push/529/

Also ensured that `./buildpkg.sh -u libkdumpfile` and `./updatelist.sh -n userland` ran as expected.

**Pull request merged**